client/java/transports-dataplex/src/main/java/io/openlineage/client/transports/dataplex/DataplexConfig.java

```diff
@@ -0,0 +1,45 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.client.transports.dataplex;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.openlineage.client.MergeConfig;
+import io.openlineage.client.transports.TransportConfig;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+public class DataplexConfig implements TransportConfig, MergeConfig<DataplexConfig> {
```

Comment:
Suggestion: we can have one more property name `mode`. which could have sync or async value. Based on these user provided value we can decide to use SyncLineageProducerClient or AsyncLineageProducerClient. This way we are giving more freedom to our user to decide how should they wanna publish lineage into Dataplex.

Reply:
@pulkit-jain-G adressed your comment, please re-review

Reply:
Thanks @ddebowczyk92!